lib/game.rb

```diff
@@ -0,0 +1,177 @@
+
+
+class Game
+
+  attr_accessor :board, :player_1, :player_2, :token, :cells, :move, :player
+ 
+
+  
+  WIN_COMBINATIONS = [[0,1,2], [3,4,5], [6,7,8], [0,3,6], [1,4,7], [2,5,8], [0,4,8], [2,4,6]]
+  def initialize(player_1 = "X", player_2 = "O", board = [" ", " ", " ", " ", " ", " ", " ", " ", " "])
+    if board == [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+      @board = Board.new
+    else
+      @board = board
+    end
+    if player_1 == "X"
+      @player_1 = Player::Human.new(player_1)
+      @player_1.board = @board
+    else
+      @player_1 = player_1
+      @player_1.board = @board
+    end
+    if player_2 == "O"
+      @player_2 = Player::Human.new(player_2)
+      @player_2.board = @board
+    else
+      @player_2 = player_2
+      @player_2.board = @board
+    end
+
+  end
+
+
+  def current_player
+    if @board.turn_count == 0
+      return @player_1
+    elsif @board.turn_count % 2 != 0
+      return @player_2
+    else
+      return @player_1
+    end
+  end
+
+  def over?
+    if !@board.cells.include?(" ") || won?
+      return true
+    else
+      return false
+    end
+  end
+  
+
+
+
+  
+  
+  def draw?
+    if !@board.cells.include?(" ") && !won?
+      true
+    else
+      false
+    end
+  end
+
+  def winner
+    if !won?
+      return nil
+    elsif @board.cells.count("X") > @board.cells.count("O")
+      return "X"
+    elsif @board.cells.count("O") >= @board.cells.count("X")
+      return "O"
+
+
+    end
+  end
+  
+
+  
+
+
+  def won?
+    status = false
+
+    Game::WIN_COMBINATIONS.each do |combo|
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.find do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]]
    end
  end
```